src/crossref.js

```
@@ -1,18 +1,272 @@
+            newItem = await this.getItemFromIdentifier({DOI: crossrefItem.DOI});
+        }
+        else if(crossrefItem.isbn){
+            newItem = await this.getItemFromIdentifier({ISBN: crossrefItem.ISBN});
+        }
+        else{
+            newItem = this.parseItemFromCrossrefReference(crossrefItem);
+        }
+        return newItem;
+    }
+
+    /**
+     * Get a Zotero Item from a valid Zotero identifier - includes DOI, ISBN, PMID, ArXiv ID, and more.
+     * @param {{string: string}} identifier - A reference item in JSON Crossref format.
+     * @returns {Promise<Zotero.Item | null>} Zotero item parsed from the identifier, or null if parsing failed.
+     */
+    static async getItemFromIdentifier(identifier){
+        await Zotero.Schema.schemaUpdatePromise;
+        let translation = new Zotero.Translate.Search();
+        translation.setIdentifier(identifier);
+
+        let jsonItems;
+        try {
+            // set libraryID to false so we don't save this item in the Zotero library
+            jsonItems = await translation.translate({libraryID: false});
+        } catch {
+            debug('No items returned for identifier ' + identifier);
+        }
+
+        if (jsonItems) {
+            const jsonItem = jsonItems[0];
+            // delete irrelevant fields to avoid warnings in Item#fromJSON
+            delete jsonItem['notes'];
+            delete jsonItem['seeAlso'];
+            delete jsonItem['attachments'];
+
+            const newItem = new Zotero.Item(jsonItem.itemType);
+            newItem.fromJSON(jsonItem);
+            return newItem;
+        }
+        else{
+            return null;
+        }
+    }
+
+    /**
+     * Get a Zotero Item from a Crossref reference item that doesn't include an identifier.
+     * @param {string} crossrefItem - A reference item in JSON Crossref format.
+     * @returns {Promise<Zotero.Item | null>} Zotero item parsed from the identifier, or null if parsing failed.
```

This doesn’t actually return a `Promise`… but it should! Return a `Promise<Zotero.Item>` (return `Promise.reject()`, rather than returning `null`).

---

Oh yeah, I missed that this wasn't async. But yeah, using `allSettled` above makes `Promise.reject()` clearer here